toolchain/check/handle_loop_statement.cpp

```diff
@@ -40,17 +67,57 @@ auto HandleForStatement(Context& context, Parse::Node parse_node) -> bool {
   return context.TODO(parse_node, "HandleForStatement");
 }
 
-auto HandleWhileCondition(Context& context, Parse::Node parse_node) -> bool {
-  return context.TODO(parse_node, "HandleWhileCondition");
-}
-
 auto HandleWhileConditionStart(Context& context, Parse::Node parse_node)
     -> bool {
-  return context.TODO(parse_node, "HandleWhileConditionStart");
+  // Branch to the loop entry block.
+  auto loop_entry_id = context.AddDominatedBlockAndBranch(parse_node);
+  context.node_block_stack().Pop();
+
+  // Start emitting the loop entry block.
+  context.node_block_stack().Push(loop_entry_id);
+  context.AddCurrentCodeBlockToFunction();
+
+  context.node_stack().Push(parse_node, loop_entry_id);
+  return true;
```

Comment:
What do you think about calling this the loop `header` block? It at least seems accurate given the LLVM terminology here:
https://llvm.org/docs/LoopTerminology.html

Notably, `entering` and `entry` are very similar but importantly different here -- the block you're adding here is inside the loop (evaluating the loop condition). Calling it the loop `header` would at least for me make this more unambiguous. TBF, that's likely because of how familiar I am with the LLVM loop terminology. But not sure there is any better terminology to anchor on here.

Reply:
Hah, I started off calling this "header" and then renamed it to "entry" to better parallel the "exit" block name. From a quick search, it looks like "header" is a substantially more popular term than "entry" outside LLVM too (though both seem to be used), and in any case, anchoring on LLVM terminology seems very reasonable here. Done :)